src/SonarScanner.MSBuild.Tasks/IsTestByReference.cs

```diff
@@ -0,0 +1,95 @@
+﻿/*
+ * SonarScanner for MSBuild
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace SonarScanner.MSBuild.Tasks
+{
+    /// <summary>
+    /// MSBuild task that determines whether the project should be treated as a test project for analysis purposes based on its references.
```

Comment:
please add a comment that this logic is duplicated and should be kept in sync with:
- sonar-dotnet
- sonar-security C# FE

Reply:
Will we keep the fallback logic in sonar-security FE actually? As it never runs in SL mode. `New plugin` + `old scanner` could produce more UCFGs, but that should not be such a big issue (?)

Reply:
Good question. I think we may still want to keep it to avoid performance regressions (as producing UCFGs takes time). But we should document that we'd keep it only for backwards compatibility with older scanners.